src/steps/active-directory/__tests__/client.test.ts

```diff
@@ -0,0 +1,154 @@
+    });
+
+    expect(resources.length).toBeGreaterThan(0);
+  });
+
+  test('inaccessible', async () => {
+    const client = new DefenderClient(logger, integrationConfig);
+
+    const resources: Machine[] = [];
+    await client.iterateFindings(
+      {
+        machineId: '1c417feb-b04f-46c9-a747-614d6d03f348',
+        select: ['id', 'displayName'],
+      },
+      (e) => {
+        resources.push();
+      },
+    );
+
+    expect(resources.length).toEqual(0);
+  });
+
+  test('insufficient permissions', async () => {
+    //This doesn't throw the correct error anymore
+
+    const client = new DefenderClient(logger, integrationConfig);
+
+    const resources: Machine[] = [];
+    await client.iterateUsers(
+      {
+        machineId: '1c417feb-b04f-46c9-a747-614d6d03f348',
+        select: ['id', 'displayName'],
+      },
+      (e) => {
+        resources.push();
+      },
+    );
+
+    expect(resources.length).toEqual(0);
+  });
+});
+
+describe('iterateUsers', () => {
+  let client: DefenderClient;
+
+  beforeEach(() => {
+    client = new DefenderClient(logger, integrationConfig);
+  });
+
+  test('single selected property', async () => {
+    recording = setupProjectRecording({
+      directory: __dirname,
+      name: 'iterateUsers_single_properties',
+    });
+
+    const resources: UserLogon[] = [];
+    await client.iterateUsers(
+      {
+        machineId: 'e76b865d4bc0c2622547459464020e9e24f51f75',
+      },
+      (e) => {
+        resources.push(e);
+      },
+    );
+  });
+
+  test('multiple selected properties', async () => {
+    recording = setupProjectRecording({
+      directory: __dirname,
+      name: 'iterateUsers_multiple_properties',
+    });
+
+    const resources: UserLogon[] = [];
+    await client.iterateUsers(
+      {
+        machineId: 'e76b865d4bc0c2622547459464020e9e24f51f75',
+      },
+      (e) => {
+        resources.push(e);
+      },
+    );
+
+    expect(resources.length).toBe(1);
+  });
+});
+
+describe('iterateFindings', () => {
+  let client: DefenderClient;
+
+  beforeEach(() => {
+    client = new DefenderClient(logger, integrationConfig);
+  });
+
+  test('single selected property', async () => {
+    recording = setupProjectRecording({
+      directory: __dirname,
+      name: 'iterateFindings_single_properties',
+    });
+
+    const resources: UserLogon[] = [];
+    await client.iterateFindings(
+      {
+        machineId: '660688d26b586b005a90cc148bfb78ed8e55b32b',
+      },
+      (e) => {
+        resources.push();
+      },
+    );
+  });
+
+  test('multiple selected properties', async () => {
+    recording = setupProjectRecording({
+      directory: __dirname,
+      name: 'iterateFindings_multiple_properties',
+    });
+
+    const resources: UserLogon[] = [];
+    await client.iterateFindings(
+      {
+        machineId: '660688d26b586b005a90cc148bfb78ed8e55b32b',
+      },
+      (e) => {
+        resources.push();
+      },
+    );
+
+    expect(resources.length).toBe(0);
```

Comment:
The `resources.push()` will never add anything, so this assertion appears meaningless.